source/shared/core_stmt.cpp

```diff
@@ -2682,6 +2661,129 @@ bool reset_ae_stream_cursor( _Inout_ sqlsrv_stmt* stmt ) {
     return false;
 }
 
+void adjustInputPrecision( _Inout_ zval* param_z, _In_ SQLSMALLINT decimal_digits ) {
+    // 38 is the maximum precision supported for sql decimal types
+    // 6 is a composition of: 1 for '.'; 1 for sign of the number;
+    //   1 for 'e' or 'E' (scientific notation); 1 for sign of scientific exponent; 2 for length of scientific exponent
+    size_t maxDecimalLen = 38 + 6;
+    SQLSRV_ASSERT( Z_STRLEN_P( param_z ) < maxDecimalLen, "Input decimal overflow: sql decimal type only supports up to a precision of 38." );
+    std::vector<size_t> digits;
+    char* ptr = ZSTR_VAL( Z_STR_P( param_z ));
+    bool isNeg = false;
+    char scientificChar = ' ';
+    int scientificExp = 0;
+    // parse digits in param_z into the vector digits
+    if( *ptr == '+' || *ptr == '-' ){
+        if( *ptr = '-' ){
+            isNeg = true;
+        }
+        ptr++;
+    }
+    size_t numInt = 0;
+    size_t numDec = 0;
+    while( isdigit( *ptr )){
+        digits.push_back( *ptr - '0' );
+        ptr++;
+        numInt++;
+    }
+    if( *ptr == '.' ){
+        ptr++;
+        while( isdigit(*ptr) ){
+            digits.push_back( *ptr - '0' );
+            ptr++;
+            numDec++;
+        }
+    }
+    if( *ptr == 'e' || *ptr == 'E' ){
+        scientificChar = *ptr;
+    }
+    if( scientificChar != ' ' ){
+        ptr++;
+        bool isNegExp = false;
+        if( *ptr == '+' || *ptr == '-' ){
+            if( *ptr == '-' ){
+                isNegExp = true;
+            }
+            ptr++;
+        }
+        while( isdigit( *ptr )){
+            scientificExp = scientificExp * 10 + *ptr - '0';
+            ptr++;
+        }
+        SQLSRV_ASSERT( scientificExp <=38, "Input decimal overflow: sql decimal type only supports up to a precision of 38." );
+        if( isNegExp ){
+            scientificExp = scientificExp * -1;
+        }
+    }
+    SQLSRV_ASSERT( *ptr == '\0', "Invalid input decimal: Invalid character found in the decimal string." );
+    // if number of decimal is less than the exponent, that means the number is a whole number, so no need to adjust the precision
+    if(( int )numDec > scientificExp ){
+        int decToRemove = numDec - scientificExp - decimal_digits;
```

Comment:
why don't you simply declare numDec as `int`? Probably  `short`  is more than enough too